ast.ts

```diff
@@ -25,6 +26,7 @@ export type Stmt<A> =
   | {  a?: A, tag: "while", cond: Expr<A>, body: Array<Stmt<A>> }
   | {  a?: A, tag: "pass" }
   | {  a?: A, tag: "field-assign", obj: Expr<A>, field: string, value: Expr<A> }
+  | {  a?: A, tag: "list_assign", list: Expr<A>, index: number, value: Expr<A> }
```

Comment:
2 points to note:
1. By using a `number` for index, you can only access/change indices known at compile time, not runtime (i.e. `myList[0 + 1] = 5` won't work).
2. Not so much an issue right now, but at some point we'll probably want to have a more generalized version of looking up the location to read/assign to for restructured assignment (e.g. `x[0], y = [1, 2]`).

Reply:
Hello from the sets-tuples-dicts group! So, we would require some sort of assign stmt as well to handle setting key-value pairs. myDict[key] = value which looks exactly similar like the list index lookup and assign statement at parse time. I propose we have something like this:
`Stmt<A>:  | {  a?: A, tag: "lookup-assign", name: string, key: Expr<A>, value: Expr<A> }`

At typecheck, the type would be resolved as dict or list and codegen can be done accordingly. How does this sound? @R167

Reply:
Yes this one should actually be `Expr<A>`, we simply forgot about it when writing the code 😅